src/sqlfe/general/Submission.java

```diff
@@ -176,60 +174,114 @@ public String toString() {
 				+ queryEvals + ", totalPoints=" + totalPoints + "]";
 	}
 	
+	// Skip comments or blank spaces before reaching a new question or an instructor comment
+	private String parseComments(String line, BufferedReader br , PrintWriter commWriter, boolean isNewQuestion ){
+		while (line != null && Utilities.isInstructorComment(line) && !isNewQuestion) {			
+		// skip remaining instructor comment lines with question text
+			line = Utilities.skipInstructorComments(br, line);
+			//System.out.println("line after skipping instructor comments with question text is: >" + line + "<");
+			
+			// skip any blank lines after instructor comments
+			line = Utilities.skipBlankLines(br, line);
+			//System.out.println("line after skipping blanks after instructor comments is: >" + line + "<");
+			
+			// process any user comments above the answer
+			line = Utilities.processUserComments(br, line, commWriter, submissionFileName);
+			//System.out.println("line after skipping user comments above answer is: >" + line + "<");
+			
+			// skip any remaining blank lines before answer
+			line = Utilities.skipBlankLines(br, line);
+			//System.out.println("line after skipping any remaining blank lines before answer is: >" + line + "<");
+		
+			// check the new line to see if is new question instructor comment
+			if (line != null && Utilities.isInstructorComment(line)) {
+		        if (Utilities.isQuestionFound(line)) {				// start of new question
+		        	isNewQuestion = true;
+		        	//System.out.println("parsing instructor comments, but found new question");
+		        } else {
+		        	//System.out.println("parsing instructor comments, found another instructor comment for same question");
+		        }
+			}
+		}	// end - while
+		return line;
+	}
+	
+	//Get the complete SQL query written by student and return the answer query string
+	public String readSQLquery(String line, String answerQueryStr, BufferedReader br,PrintWriter commWriter)
+	{
+		//start of answer (possibly complete on one line) unless no answer present, then make answerQueryStr blank
+		if (line != null && !Utilities.isInstructorComment(line)) {
+			answerQueryStr = line;
+		}else if (Utilities.isInstructorComment(line)) {	// if found next question - no answer submitted
+			answerQueryStr = "";
+			return answerQueryStr;
+		}else {
+			answerQueryStr = "";
+		} 
+		//System.out.println("start of answerQueryStr is: >" + answerQueryStr + "<");
+		
+		// process the remaining lines for that answer to get the complete query
+		try {
+			while((line = br.readLine())!= null) {
+				
+				// if not at end of file...
+			    // TODO - need to generally check for . or )
+			    // TODO - need better new question check than period at less than hard-coded position
+			    if (Utilities.isQuestionFound(line) && line.indexOf('.') < 8) {	// if find start of next question
```

Comment:
Do we need the period index check here? If so, should this be part of Utilities.isQuestionFound method?

Reply:
I think we dont need the index check here. I removed it and ran the app, the files were parsed and evaluated correctly.

Reply:
Awesome